feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonCache.kt

```diff
@@ -0,0 +1,45 @@
+package com.revenuecat.purchases.amazon
+
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.caching.DeviceCache
+import com.revenuecat.purchases.common.log
+import org.json.JSONObject
+
+const val RECEIPTS_TO_SKUS_KEY = "receiptsToSkus"
+
+internal class AmazonCache(
+    private val deviceCache: DeviceCache
+) {
+
+    internal val amazonPostedTokensKey: String by lazy {
+        deviceCache.newKey("amazon.tokens")
+    }
+
+    @Synchronized
+    fun setReceiptSkus(receiptsToSkus: Map<String, String>) {
+        log(LogIntent.DEBUG, AmazonStrings.CACHING_RECEIPT_TERM_SKUS.format(receiptsToSkus))
+
+        val currentlyCached = getReceiptSkus()
+
+        val toCache = currentlyCached + receiptsToSkus
+
+        val innerJSONObject = JSONObject(toCache)
+        val jsonToCache = JSONObject().also { it.put(RECEIPTS_TO_SKUS_KEY, innerJSONObject) }
+        deviceCache.putString(amazonPostedTokensKey, jsonToCache.toString())
+    }
+
+    @Synchronized
+    fun getReceiptSkus(): Map<String, String> {
+        val receiptToSkusJSONObject =
+            deviceCache.getJSONObjectOrNull(amazonPostedTokensKey)?.getJSONObject(RECEIPTS_TO_SKUS_KEY)
+
+        return receiptToSkusJSONObject?.keys()?.asSequence()?.map { jsonKey ->
+            jsonKey to receiptToSkusJSONObject[jsonKey] as String
+        }?.toMap() ?: emptyMap()
```

Comment:
there isn't a more direct way of converting a JSONObject to map?

Reply:
Not that I can think of. Would it help extracting it to a function?

Reply:
if we do it often, then moving to a function would be a good idea. 
idea for a very small improvement: `.keys()?.asSequence()` -> `.keySet()`

Reply:
`keySet` is hidden and not accessible. I will extract it to a util function